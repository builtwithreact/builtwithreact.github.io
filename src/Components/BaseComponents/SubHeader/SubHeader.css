.Heading-Container {
  height: 200px;
  background: #1c1d21;
  width: 100%;
  color: #fff; }
  .Heading-Container .Heading {
    text-align: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: 100%; }
  .Heading-Container .Tagline {
    font-size: 30px; }
  .Heading-Container .Sub-Tagline {
    font-size: 25px; }
