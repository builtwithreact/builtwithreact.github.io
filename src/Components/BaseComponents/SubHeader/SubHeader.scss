.Heading-Container {
  height: 200px;
  background: #1c1d21;
  width: 100%;
  color: #fff;
  .Heading {
    text-align: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: 100%;
  }
  .Tagline {
    font-size: 30px;
  }
  .Sub-Tagline {
    font-size: 25px;
  }
}