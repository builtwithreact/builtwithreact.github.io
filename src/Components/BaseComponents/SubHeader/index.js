import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import './SubHeader.css';

class SubHeader extends Component {
    constructor(props) {
        super(props);
        this.state = {
            isWordVisible: true
        }
        this.CONSTANTS = {
            SUBMIT_PROJECT_BTN_TITLE: "Submit Request",
            GIVE_FEEDBACK_BTN_TITLE: "Give Feedback"
        };

        setInterval(() => {
            this.setState({
                isWordVisible: !this.state.isWordVisible
            });
        }, 2500)

    }

    render() {
        return (
            <div className="HeadingContainer">
                <div className="Heading container">
                    <h1 className="Tagline">One place to showcase your work</h1>
                    <div className="SubTagline">
                        <span>Implemented </span>
                        <span className={`SubTagline__Item ${this.state.isWordVisible ? 'SubTagline__Item--Front' : 'SubTagline__Item--Back'}`}>
                            <span className={this.state.isWordVisible ? 'is-visible' : 'is-hidden'}>using</span>
                            <span className={!this.state.isWordVisible ? 'is-visible' : 'is-hidden'}>for</span>
                        </span>
                        <span> React / React Native</span>
                    </div>
                    <div className="Heading__Cta">
                        <Link to="/SubmitRequest" className="Button">{this.CONSTANTS.SUBMIT_PROJECT_BTN_TITLE}</Link>
                        <Link to="/Feedback" className="Button">{this.CONSTANTS.GIVE_FEEDBACK_BTN_TITLE}</Link>
                    </div>
                </div>
            </div>
        );
    }
}


export default SubHeader;