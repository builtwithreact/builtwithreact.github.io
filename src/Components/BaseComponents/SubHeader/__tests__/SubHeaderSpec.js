import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom'
import SubHeader from './../index';

describe('SubHeader', () => {
    it('renders correctly', () => {
        let tree = renderer.create(
            <StaticRouter location="someLocation" context={{}}>
                <SubHeader />
            </StaticRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
