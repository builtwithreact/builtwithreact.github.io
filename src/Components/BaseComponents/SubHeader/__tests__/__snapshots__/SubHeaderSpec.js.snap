// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubHeader renders correctly 1`] = `
<div
  className="HeadingContainer"
>
  <div
    className="Heading container"
  >
    <h1
      className="Tagline"
    >
      One place to showcase your work
    </h1>
    <div
      className="SubTagline"
    >
      <span>
        Implemented 
      </span>
      <span
        className="SubTagline__Item SubTagline__Item--Front"
      >
        <span
          className="is-visible"
        >
          using
        </span>
        <span
          className="is-hidden"
        >
          for
        </span>
      </span>
      <span>
         React / React Native
      </span>
    </div>
    <div
      className="Heading__Cta"
    >
      <button
        className="Button"
        onClick={[Function]}
        type="button"
      >
        Submit Request
      </button>
      <button
        className="Button"
        onClick={[Function]}
        type="button"
      >
        Give Feedback
      </button>
    </div>
  </div>
</div>
`;
